src/overlays/actors/ovl_En_Dha/z_en_dha.c

```diff
@@ -315,54 +299,44 @@ void EnDha_UpdateHealth(EnDha* this, GlobalContext* globalCtx) {
                 this->actor.colChkInfo.health = 8;
                 Item_DropCollectibleRandom(globalCtx, &this->actor, &this->actor.posRot.pos, 0xE0);
             } else {
-                Audio_PlayActorSound2(this, NA_SE_EN_DEADHAND_DAMAGE);
+                Audio_PlayActorSound2(&this->actor, NA_SE_EN_DEADHAND_DAMAGE);
                 this->unk_1C0 = 9;
                 func_809ECF60(this);
             }
         }
     }
-    if (this->actor.parent != NULL) {
-        if (this->actor.parent->params == 0xA) {
-            EnDha_SetupDeath(this);
-        }
+    if ((this->actor.parent != NULL)  && (this->actor.parent->params == ENDH_DEATH)) {
+        EnDha_SetupDeath(this);
     }
 }
 
 void EnDha_Update(Actor* thisx, GlobalContext* globalCtx) {
+    s32 pad;
     EnDha* this = THIS;
-    CollisionCheckContext* colChkCtx;
-
-    colChkCtx = &globalCtx->colChkCtx;
 
     if (this->actor.parent == NULL) {
         this->actor.parent = Actor_FindNearby(globalCtx, &this->actor, ACTOR_EN_DH, ACTORTYPE_ENEMY, 10000.0f);
     }
     EnDha_UpdateHealth(this, globalCtx);
     this->actionFunc(this, globalCtx);
-    CollisionCheck_SetAC(globalCtx, colChkCtx, &this->collider);
-    CollisionCheck_SetOC(globalCtx, colChkCtx, &this->collider);
+    CollisionCheck_SetAC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
+    CollisionCheck_SetOC(globalCtx, &globalCtx->colChkCtx, &this->collider.base);
 }
 
 s32 EnDha_OverrideLimbDraw(GlobalContext* globalCtx, s32 limbIndex, Gfx** dList, Vec3f* pos, Vec3s* rot, Actor* thisx) {
     EnDha* this = THIS;
-    s32 retVar;
 
     if (limbIndex == 1) {
         rot->y = -(s16)(this->unk_1CE + 0x4000);
         rot->z += this->unk_1D0.y;
-        retVar = 0;
     } else if (limbIndex == 2) {
         rot->z = this->unk_1D0.x;
         rot->y -= this->unk_1D0.y;
-        retVar = 0;
     } else if (limbIndex == 3) {
-        rot->y = -(this->unk_1D6.x);
-        rot->z = -(this->unk_1D0.z);
-        retVar = 0;
-    } else {
-        retVar = 0;
+        rot->y = -this->unk_1D6.x;
+        rot->z = -this->unk_1D0.z;
     }
```

Comment:
Do we know what these indices match up to? Could we replace those numbers with an enum or #define?

Reply:
Probably, but we haven't done that thing a lot in the past. Also I didn't decomp this actor--I just cleaned it up some because it references En_Dh.

Reply:
yeah thats certainly more of a documentation step. if people want to identify limbs and name them they can, but its not required